daemon/api.go

```diff
@@ -789,14 +789,18 @@ func snapUpdateMany(inst *snapInstruction, st *state.State) (msg string, updated
 		return "", nil, nil, err
 	}
 
-	switch len(inst.Snaps) {
+	switch len(updated) {
 	case 0:
-		// all snaps
-		msg = i18n.G("Refresh all snaps in the system")
+		// not really needed but be paranoid
+		if len(inst.Snaps) != 0 {
+			return "", nil, nil, fmt.Errorf("internal error, when asking for a refresh of %q no update was found but not error was generated", inst.Snaps)
```

Comment:
...but *no* error was generated. (~t~)

Also: can you make it `internal error:`, with a colon instead of a comma? We're using a comma in 5 such places (and they're all wrong :-) ), vs 95 colons.